Finjector.Web/ClientApp/src/pages/Landing.test.tsx

```diff
@@ -34,9 +63,130 @@ describe("Landing", () => {
     await waitFor(() => {
       expect(screen.getByText("Chart 1")).toBeInTheDocument();
     });
+
+    await waitFor(() => {
+      expect(screen.getByText("Team 0")).toBeInTheDocument();
+    });
+
+    await waitFor(() => {
+      expect(screen.getByText("Folder 0")).toBeInTheDocument();
+    });
+  });
+
+  it("Example to override what the mock is returning", async () => {
+    //override the mocked data to set specific values for the team and folder
+    server.use(
+      http.get("/api/charts/all", () =>
+        HttpResponse.json([
+          {
+            team: fakeTeams[2],
+            folders: [fakeFolders[2]],
+          },
+        ])
+      )
+    );
+    // render component
+    render(wrappedView());
+
+    // should see list of saved charts (3 from our mock data)
+    await waitFor(() => {
+      expect(screen.getByText("Chart 1")).toBeInTheDocument();
+    });
+
+    await waitFor(() => {
+      expect(screen.getByText("Team 2")).toBeInTheDocument();
+    });
+
+    await waitFor(() => {
+      expect(screen.getByText("Folder 2")).toBeInTheDocument();
+    });
+
+    //console.log(screen.debug());
   });
-});
 
+  //Test that the search/filter field works
+  it("searches for a chart", async () => {
+    const user = userEvent.setup(); // at the top of the test
+
+    // render component
+    render(wrappedView());
+
+    // should see list of saved charts (3 from our mock data)
+    await waitFor(() => {
+      expect(screen.getByText("Chart 1")).toBeInTheDocument();
+    });
+    await waitFor(() => {
+      expect(screen.getByText("Chart 2")).toBeInTheDocument();
+    });
+
+    // search for a chart
+    const searchField = screen.getByPlaceholderText(
+      "Search my chart strings, teams and folders"
+    ) as HTMLInputElement;
+
+    //type text into the search field
+    await user.type(searchField, "Chart 2");
+
+    // should only see the chart we searched for
+    await waitFor(() => {
+      expect(screen.getByText("Chart 2")).toBeInTheDocument();
+      //Expect that the other chart is not present
+      expect(screen.queryByText("Chart 1")).not.toBeInTheDocument();
+    });
+    //show what the screen looks like
+    //console.log(screen.debug());
+  });
+
+  it("expects details and use buttons", async () => {
+    // Mock window.opener
+    Object.defineProperty(window, "opener", {
+      writable: true,
+      value: {}, // non-null object makes window.opener truthy
+    });
+
+    const user = userEvent.setup(); // at the top of the test
+    // render component
+    render(wrappedView());
+
+    // search for a chart
+    const searchField = screen.getByPlaceholderText(
+      "Search my chart strings, teams and folders"
+    ) as HTMLInputElement;
```

Comment:
eventually would be nice to not search by text -- use label or name instead.  we'd want to be able to change the placeholder w/o causing tests to fail